elasticsearch/os_client_v1.go

```diff
@@ -259,3 +259,109 @@ func (os *OSClientV1) GetDBUserRole(ctx context.Context) (error, bool) {
 func (os *OSClientV1) CreateDBUserRole(ctx context.Context) error {
 	return errors.New("not supported in os version 1")
 }
+
+func (os *OSClientV1) IndexExistsOrNot(_index string) (bool, error) {
+	req := opensearchapi.IndicesExistsRequest{
+		Index: []string{_index},
+	}
+	res, err := req.Do(context.Background(), os.client)
+	if err != nil {
+		return false, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode > 299 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (os *OSClientV1) CreateIndex(_index string) error {
+	reqCreateIndex := opensearchapi.IndicesCreateRequest{
+		Index:  _index,
+		Pretty: true,
+		Human:  true,
+	}
+
+	res, err := reqCreateIndex.Do(context.Background(), os.client)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		return decodeError(res.Body, res.StatusCode)
+	}
+
+	return nil
+}
+
+func (os *OSClientV1) CountData(_index string) (int, error) {
+	req := opensearchapi.CountRequest{
+		Index: []string{_index},
+	}
+
+	res, err := req.Do(context.Background(), os.client)
+	if err != nil {
+		return 0, err
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		return 0, decodeError(res.Body, res.StatusCode)
+	}
+
+	var response map[string]interface{}
+	if err := json.NewDecoder(res.Body).Decode(&response); err != nil {
+		return 0, err
+	}
+
+	count := int(response["count"].(float64))
```

Comment:
check if count exists in response body
count , ok := response["count"]
if !ok {
	return 0, errors.new("Failed to parse value for index count in response body")
}

return count , nil